staging/src/k8s.io/code-generator/cmd/applyconfiguration-gen/generators/applyconfiguration.go

```diff
@@ -242,14 +242,26 @@ func (g *applyConfigurationGenerator) generateMemberWith(sw *generator.SnippetWr
 	sw.Do("// and returns the receiver, so that objects can be built by chaining \"With\" function invocations.\n", memberParams)
 	sw.Do("// If called multiple times, the $.Member.Name$ field is set to the value of the last call.\n", memberParams)
 	sw.Do("func (b *$.ApplyConfig.ApplyConfiguration|public$) With$.Member.Name$(value $.MemberType|raw$) *$.ApplyConfig.ApplyConfiguration|public$ {\n", memberParams)
-	g.ensureEnbedExistsIfApplicable(sw, memberParams)
+	g.ensureEmbedExistsIfApplicable(sw, memberParams)
 	if g.refGraph.isApplyConfig(memberParams.Member.Type) || isNillable(memberParams.Member.Type) {
 		sw.Do("b.$.Member.Name$ = value\n", memberParams)
 	} else {
 		sw.Do("b.$.Member.Name$ = &value\n", memberParams)
 	}
 	sw.Do("  return b\n", memberParams)
 	sw.Do("}\n", memberParams)
+
+	if memberParams.Member.Name == "Name" {
```

Comment:
I don’t like having a special case here. I haven’t been able to find a way to avoid needing a new method here to retrieve the `.Name` field, but if there is a way that would be better. Otherwise, does it make sense to add a getter for all `With` fields?

Reply:
Is this comment still relevant or is it obsolete with the latest commits?  I wasn't able to locate it.

Reply:
It moved here, https://github.com/kubernetes/kubernetes/pull/121439/files#diff-4e41628987ab75d9b73ad2d8633eb6d0b2f056e98c0ae854ada75baa5e2fe226R147

```go
func needsGetter(t *types.Type, member types.Member) bool {
	// Needed when applying an ApplyConfiguration
return objectMeta.Name == t.Name && member.Name == "Name"
}

func (g *applyConfigurationGenerator) generateGetters(t *types.Type, typeParams TypeParams, sw *generator.SnippetWriter, embed *memberParams) {
	for _, member := range t.Members {
		if blocklisted(t, member) {
			continue
		}
		memberType := g.refGraph.applyConfigForType(member.Type)
		if g.refGraph.isApplyConfig(member.Type) {
			memberType = &types.Type{Kind: types.Pointer, Elem: memberType}
		}
		if jsonTags, ok := lookupJSONTags(member); ok {
			memberParams := memberParams{
				TypeParams: typeParams,
				Member:     member,
				MemberType: memberType,
				JSONTags:   jsonTags,
				EmbeddedIn: embed,
			}
			if memberParams.Member.Embedded {
				g.generateGetters(member.Type, typeParams, sw, &memberParams)
				continue
			}

			if needsGetter(t, member) {
				g.generateMemberGetter(sw, memberParams)
			}
		}
	}
}
```